frontend/src/metabase/browse/components/utils.tsx

```diff
@@ -37,3 +45,55 @@ export const getModelDescription = (item: SearchResult) => {
     return item.description;
   }
 };
+
+export const getCollectionPathString = (collection: CollectionEssentials) => {
+  const ancestors: CollectionEssentials[] =
+    collection.effective_ancestors || [];
+  const collections = ancestors.concat(collection);
+  const pathString = collections
+    .map(coll => getCollectionName(coll))
+    .join(` ${pathSeparatorChar} `);
+  return pathString;
+};
+
+const getValueForSorting = (
+  model: ModelResult,
+  sort_column: keyof ModelResult,
+): string => {
+  if (sort_column === "collection") {
+    return getCollectionPathString(model.collection);
+  } else {
+    return model[sort_column];
+  }
+};
+
+export const isValidSortColumn = (
+  sort_column: string,
+): sort_column is keyof ModelResult => {
+  return ["name", "collection"].includes(sort_column);
+};
+
+export const sortModels = (
+  models: ModelResult[],
+  sortingOptions: SortingOptions,
+  localeCode: string = "en",
+) => {
+  const { sort_column, sort_direction } = sortingOptions;
+
+  if (!isValidSortColumn(sort_column)) {
+    console.error("Invalid sort column", sort_column);
+    return models;
+  }
+
+  return [...models].sort((a, b) => {
+    const aValue = getValueForSorting(a, sort_column);
+    const bValue = getValueForSorting(b, sort_column);
```

Comment:
Could we save this for the performance follow-up PR?

I'm also not yet sure if this will help. I think we'll need to JSON.stringify the deps, which is about as expensive as running the function itself.

Reply:
I think it's fine to do this in a follow-up perf PR. 👍 Not sure what the actual runtime impact is.